vyper/codegen/expr.py

```diff
@@ -536,6 +536,14 @@ def parse_BinOp(self):
                     pos=pos,
                 )
 
+        # If the types do not match, check if type definition is propagated in node's metadata.
+        # This step is necessary for struct members.
+        if left.typ.typ != right.typ.typ:
+            if hasattr(self.expr.left, "_metadata") and "type" in self.expr.left._metadata:
```

Comment:
there is no need to be so gentle here, we can go directly for `self.expr.left._metadata["type"]`. if the field is not there, it's a typechecker bug and we should fix it